prov/rxm/src/rxm_cq.c

```diff
@@ -1256,7 +1256,9 @@ static ssize_t rxm_handle_atomic_req(struct rxm_ep *rxm_ep,
 		void *src_buf = req_hdr->data + offset;
 		void *cmp_buf = req_hdr->data + len + offset;
 		void *res_buf = resp_hdr->data + offset;
-		void *dst_buf = (void *) req_hdr->rma_ioc[i].addr;
+		void *dst_buf;
+		uintptr_t addr = (uintptr_t) req_hdr->rma_ioc[i].addr;
+		dst_buf = (void *) addr;
```

Comment:
no need to define new variable. Just change the original typecast to `(void *)(uintptr_t)`